src/org/intellij/erlang/debugger/xdebug/ErlangXDebugProcess.java

```diff
@@ -177,9 +212,18 @@ public void breakpointIsSet(String module, int line) {
   }
 
   @Override
-  public void breakpointReached(final OtpErlangPid pid, List<ErlangProcessSnapshot> snapshots) {
+  public synchronized void breakpointReached(final OtpErlangPid pid, List<ErlangProcessSnapshot> snapshots) {
     ErlangProcessSnapshot processInBreakpoint = ContainerUtil.find(snapshots, erlangProcessSnapshot -> erlangProcessSnapshot.getPid().equals(pid));
     assert processInBreakpoint != null;
+
+    boolean ignoreIfSame = myIgnoreNextBreakpointIfSame;
+    myIgnoreNextBreakpointIfSame = false;
```

Comment:
A bit tricky assignments, do you need that because of the syncronization?

Reply:
See comment above :)